Plots/Station/NCL_station_3.py

```diff
@@ -0,0 +1,157 @@
+    fig = plt.figure(figsize=(12, 6.5))
+
+    # Generate axes
+    ax = plt.axes(projection=ccrs.Mercator())
+
+    # Set extent to show particular area of the map
+    ax.set_extent([25.5, 45.2, 35.5, 42.5], ccrs.PlateCarree())
+
+    # Add state boundaries other lake features
+    ax.add_feature(cfeature.LAND, facecolor='none', edgecolor='gray')
+
+    # Draw gridlines
+    gl = ax.gridlines(crs=ccrs.PlateCarree(),
+                      draw_labels=True,
+                      dms=False,
+                      x_inline=False,
+                      y_inline=False,
+                      linewidth=1,
+                      color="black",
+                      alpha=0.25)
+
+    # Set frequency of gridlines in the x and y directions
+    gl.xlocator = mticker.FixedLocator(np.arange(26, 45, 2))
+    gl.ylocator = mticker.FixedLocator(np.arange(36, 43, 1))
+
+    # Turn off gridlines and top/right labels
+    gl.xlines = False
+    gl.ylines = False
+    gl.top_labels = False
+    gl.right_labels = False
+
+    # Set label sizes
+    gl.xlabel_style = {"rotation": 0, "size": 14}
+    gl.ylabel_style = {"rotation": 0, "size": 14}
+
+    # Manually turn off ticks on top and right spines
+    ax.tick_params(axis='x', top=False)
+    ax.tick_params(axis='y', right=False)
+
+    # Add title
+    ax.set_title(maintitle, fontweight='bold', fontsize=18, y=1.03)
+
+    return fig, ax
+
+
+##############################################################################
+# Plot with texts overlapping
+
+fig, ax = create_axes('Overlapping text strings')
+
+# Add all station number texts
+for i in range(npts):
+    ax.text(lon[i],
+            lat[i],
+            no[i],
+            fontsize=8,
+            fontweight='bold',
+            va='center',
+            ha='center',
+            transform=ccrs.PlateCarree())
+
+# Show the plot
+plt.tight_layout()
+plt.show()
+
+##############################################################################
+# Plot without texts overlapping
+
+fig, ax = create_axes('Overlapping text strings removed')
+
+# Transpose the array of longitude and latitude for easier access of the location of each station point
+location = np.transpose(np.array([lon, lat]))
+
+# Transform location from user data coodinate system into pixel coordinate system
+# Distance between two stations on screen can be more accurately calculated
+location = ax.transData.transform(location)
+
+# Create an array of booleans denoting if station would be removed
+remove = np.full(npts, False)
+
+# Currently minimum distance is calcualted through finding distance between two suitable stations
+# In the future we would like to find mindist by finding the width and height of texts in pixel coodinates
+mindist = np.sqrt(np.sum(np.square(location[123] - location[124])))
+
+# Tag station to be removed using array `remove`
+# Loop through every pair of stations and calculate distances between them
+for i in range(npts):
```

Comment:
I'm not sure this is the best way to do this, but I cannot think of a better solution, @michaelavs any ideas?

Reply:
My only other thought would be maybe an if else statement, but I also don't know if that would necessarily be more efficient or not.

Reply:
I'm wondering, if we have to compare each point against every other point then the runtime has to be O(N*2) or am I missing something?